optimizely/helpers/event_tag_utils.py

```diff
@@ -19,28 +19,69 @@
 NUMERIC_METRIC_TYPE = 'value'
 
 
-def get_revenue_value(event_tags):
+def get_revenue_value(event_tags, logger):
+  """
+  A smart getter of the revenue value from the event tags.
+
+  Args:
+    event_tags: A dictionary of event tags.
+    logger: Optional logger.
+
+  Returns:
+    An integer revenue value is returned when the provided revenue
+    value is in the following format:
+      - A string (properly formatted, e.g., no commas)
+      - An integer
+      - A float or double
+      - 4.0 or "4.0" will be parsed to int(4).
+    None is returned when the provided revenue values is in
+    the following format:
+      - None
+      - A boolean
+      - inf, -inf, nan
+      - A string not properly formatted (e.g., '1,234')
+      - 4.1 can not be parsed
+      - Any values that cannot be cast to a float (e.g., an array or dictionary)
+  """
+
   if event_tags is None:
+    logger.log(enums.LogLevels.DEBUG, 'Event tags is undefined.')
     return None
 
   if not isinstance(event_tags, dict):
+    logger.log(enums.LogLevels.DEBUG, 'Event tags is not a hash.')
     return None
 
   if REVENUE_METRIC_TYPE not in event_tags:
+    logger.log(enums.LogLevels.DEBUG, 'The revenue key is not defined in the event tags.')
     return None
 
   raw_value = event_tags[REVENUE_METRIC_TYPE]
 
   if isinstance(raw_value, bool):
+    logger.log(enums.LogLevels.WARNING, 'Revenue value is boolean.')
```

Comment:
You can eliminate boolean condition with this below:
if type(whatever) in (float, int):
    # Do stuff.